crypto-ffi/src/generic.rs

```diff
@@ -539,4 +541,27 @@ impl CoreCrypto<'_> {
 
         Ok(())
     }
+
+    // Preserves "current" behaviour with auto-merged commits
+    // TODO: remove when backend counterpart implemented
+    #[allow(clippy::await_holding_lock)]
+    async fn maybe_accept_commit<T>(
```

Comment:
No async methods in this file; even if they are internal, they might deadlock because you're synchronously locking a `Mutex` then awaiting on it, probably locking it for `'static`